ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -151,8 +156,10 @@ export default class PermissionConnect extends Component {
       history.replace(DEFAULT_ROUTE);
       return;
     }
+
     // if this is an incremental permission request for permitted chains, skip the account selection
     if (
+      // TODO pretty sure this is not needed anymore.
```

Comment:
At the very least this should probably be referencing the new permission (if at all)

Reply:
The ApprovalController still receives the old `eth_accounts` and `endowment:permittedChains` in it's props, so we can't remove them quite yet

Reply:
Well, if we aren't using `requestPermissionsIncremental` anymore then we can remove this, right?

Reply:
Yeah I suppose so. Just wondering if anyone else has any objections here incase I've missed a use case/entrypoint

Reply:
This is for the `wallet_switchEthereumChain` case where we're prompting user to grant permissions for a network for which they don't yet have a chainpermission granted. Will check if removing this breaks that flow.

Reply:
Ahh sorry, I'm wrong. We are still using the incremental view when requesting new permitted chains, but we only use the ApprovalController part of the flow and then manually grant via PermissionController. This is still needed

Reply:
removed comment here https://github.com/MetaMask/metamask-extension/pull/27847/commits/8ac50a4d5cb2ea50937cc612ad701a8324764720

Reply:
How is a permission diff generated if the permissions don't exist anymore? Are we manually creating the diff somewhere?

Reply:
You're right... I may have jumped to an incorrect conclusion here. Let me revisit tomorrow with a clearer head

Reply:
Thank you for being persistent here. I can't see how this path can be triggered anymore and have removed it.

https://github.com/MetaMask/metamask-extension/pull/27847/commits/eb0876123481f6cf2f5a9a956a18a3d08d8cf1e4